x-pack/plugins/cases/server/client/metrics/types.ts

```diff
@@ -10,4 +10,5 @@ import { CaseMetricsResponse } from '../../../common/api';
 export interface MetricsHandler {
   getFeatures(): Set<string>;
   compute(): Promise<CaseMetricsResponse>;
+  setupFeature?(feature: string): void;
```

Comment:
Allows the handler to optionally implement a method to do some pre-compute setup for each feature.